ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_composite_op.cpp

```diff
@@ -116,20 +115,20 @@ Tensor _atanh(const Tensor& input_a, const std::optional<MemoryConfig>& output_m
    {
        Tensor nr_term(input_a);
        {
-           Tensor pos_x = ttnn::add(input_a, 1.0f);
-           Tensor neg_x = ttnn::subtract(input_a, 1.0f);
+           Tensor pos_x = ttnn::add(input_a, 1.0f, std::nullopt, output_mem_config);
+           Tensor neg_x = ttnn::subtract(input_a, 1.0f, std::nullopt, output_mem_config);
            nr_term = ttnn::log(
                ttnn::multiply(
                    pos_x, ttnn::reciprocal(ttnn::neg(neg_x, output_mem_config), output_mem_config), std::nullopt, output_mem_config),
                output_mem_config);
        }
-       comp_result = ttnn::multiply(nr_term, 0.5f);
+       comp_result = ttnn::multiply(nr_term, 0.5f, std::nullopt, output_mem_config);
    }
    // Input is -1 > value > 1, output is nan
    // Input is -1 < value < 1, output is atanh(input)
    float t_nan = std::nanf("");
-   Tensor abs_temp = ttnn::subtract(ttnn::abs(input_a, output_mem_config), 1.0f);
-   Tensor result = where(ttnn::ltz(abs_temp, output_mem_config), comp_result, t_nan);
+   Tensor abs_temp = ttnn::subtract(ttnn::abs(input_a, output_mem_config), 1.0f, std::nullopt, output_mem_config);
```

Comment:
what is the logic? why do we sometimes pass output_mem_config and sometimes not?